### ISSUE TITLE
Compile failure w/ template-haskell-2.11

### ISSUE THREAD

**Issue by hvr** [MedeaMelana/Piso#1]
```
In order, the following will be built (use -v for more details):
 - Piso-0.1 {Piso-0.1-inplace} (lib:Piso) (first run)
Configuring Piso-0.1...
Preprocessing library Piso-0.1...
[1 of 4] Compiling Data.Piso        ( Data/Piso.hs, /tmp/matrix-worker/1493986712/dist-newstyle/build/x86_64-linux/ghc-8.0.2/Piso-0.1/build/Data/Piso.o )
[2 of 4] Compiling Data.Piso.Generic ( Data/Piso/Generic.hs, /tmp/matrix-worker/1493986712/dist-newstyle/build/x86_64-linux/ghc-8.0.2/Piso-0.1/build/Data/Piso/Generic.o )
[3 of 4] Compiling Data.Piso.TH     ( Data/Piso/TH.hs, /tmp/matrix-worker/1493986712/dist-newstyle/build/x86_64-linux/ghc-8.0.2/Piso-0.1/build/Data/Piso/TH.o )

Data/Piso/TH.hs:28:15: error:
    • The constructor ‘DataD’ should have 6 arguments, but has been given 5
    • In the pattern: DataD _ _ _ cons _
      In the pattern: TyConI (DataD _ _ _ cons _)
      In a case alternative:
          TyConI (DataD _ _ _ cons _)
            -> mapM (derivePiso (length cons /= 1)) cons
```

I've already updated the meta-data on Hackage to exclude configurations known to fail:

 - https://hackage.haskell.org/package/Piso-0.1/revisions/

**Comment by MedeaMelana** [comment]
Thanks for reporting! I've uploaded v 0.2 which should work with GHC 8. Thanks again :-)

**Comment by MedeaMelana** [comment]
Fixed at commit 2733909
